### PR TITLE
Move k9mail-library to Timber

### PR DESCRIPTION
**Pull request by philipwhiuk**

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/internet/MessageExtractor.java [hunk 1]
```diff
@@ -62,9 +61,9 @@ public static String getTextFromPart(Part part, long textSizeLimit) {
                 throw new MessagingException("Provided invalid part");
             }
         } catch (IOException e) {
-            Log.e(LOG_TAG, "Unable to getTextFromPart", e);
+            Timber.e(e, "Unable to getTextFromPart");
         } catch (MessagingException e) {
-            Log.e(LOG_TAG, "Unable to getTextFromPart", e);
+            Timber.e("Unable to getTextFromPart");
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/internet/MessageExtractor.java]: missing e

File: k9mail-library/src/main/java/com/fsck/k9/mail/power/TracingPowerManager.java [hunk 2]
```diff
@@ -25,7 +23,7 @@ public static synchronized TracingPowerManager getPowerManager(Context context)
         Context appContext = context.getApplicationContext();
         if (tracingPowerManager == null) {
             if (K9MailLib.isDebug()) {
```
Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/power/TracingPowerManager.java]: I think the conditionals are no longer necessary

Comment (philipwhiuk) [inline k9mail-library/src/main/java/com/fsck/k9/mail/power/TracingPowerManager.java]: Hmm. Those conditionals relate to properties in K-9. Should we remove those too? @cketti ?

Comment (Valodim) [inline k9mail-library/src/main/java/com/fsck/k9/mail/power/TracingPowerManager.java]: The way we did it in k9 itself was to plant/uproot the log tree in the places where the setting is set (and on startup).

**Comment by cketti** [comment]
At some point I'd like k9mail-library to be a Java-only library that can easily be used by third parties. To achieve that goal using something like slf4j is preferable to a dependency on Timber (which in turn depends on Android).
However, it's not an immediate goal and it's probably not worth the hassle right now.

It looks like `K9MailLib.isDebug()` is used exclusively to guard log statements. I think we can get rid of that, together with the `DEBUG_PROTOCOL_*` constants. But we do want to keep `K9MailLib.isDebugSensitive()`.

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java [hunk 3]
```diff
@@ -959,22 +956,21 @@ DataSet processRequest(String url, String method, String messageBody, Map<String
 
                     dataset = myHandler.getDataSet();
                 } catch (SAXException se) {
-                    Log.e(LOG_TAG,
-                            "SAXException in processRequest() " + se + "\nTrace: " + WebDavUtils.processException(se));
+                    Timber.e(se,
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java]: This can be a single line now.

**Comment by Valodim** [comment]
rebased and fixed in https://github.com/k9mail/k-9/tree/timberLibrary

**Comment by philipwhiuk** [comment]
You can't? That sucks. I thought i enabled that.

**Comment by Valodim** [comment]
see #2423

**Pull request closed without merging**